bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/driver/Port.java

```diff
@@ -154,7 +152,8 @@ public void clearModemDB() {
         logger.debug("clearing modem db!");
         Map<InsteonAddress, @Nullable ModemDBEntry> dbes = getDriver().lockModemDBEntries();
         for (InsteonAddress addr : dbes.keySet()) {
-            if (!dbes.get(addr).isModem()) {
+            ModemDBEntry dbe = dbes.get(addr);
+            if (dbe != null && !dbe.isModem()) {
```

Comment:
The proper fix is here to remove `@Nullable` from the dbes value type. Then if you iterate over the `entrySet` instead of the `keySet` there will be no possible null reference and so you won't need the `dbe != null` check either. It results in less code, less NPEs and better performance. :smile:

Reply:
fixed

Reply:
The code is still iterating over the `keySet` instead of the `entrySet` causing redundant null checks. Looks like the same improvement can also be applied to other for loops in the binding.

Reply:
sorry, fixed now. I also review reviewed other for-loops and didn't find any that should be changed.

Reply:
sorry, fixed now. I also review reviewed other for-loops and didn't find any that should be changed.